protostar-rust/src/cheatcodes_hint_processor.rs

```diff
@@ -365,23 +368,51 @@ fn execute_cheatcode_hint(
-            let return_data = tx_result
-                .execute_call_info
-                .expect("Failed to get execution data from method")
-                .execution
-                .retdata;
-            let contract_address = return_data
-                .0
-                .get(0)
-                .expect("Failed to get contract_address from return_data");
-            let contract_address = Felt252::from_bytes_be(contract_address.bytes());
-
-            insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
-            // todo in case of error, consider filling the panic data instead of packing in rust
-            insert_value_to_cellref!(vm, panic_data_start, Felt252::from(0))?;
-            insert_value_to_cellref!(vm, panic_data_end, Felt252::from(0))?;
-
+            match account_tx.execute(blockifier_state, block_context) {
+                Result::Ok(tx_info) => {
+                    let return_data = tx_info
+                        .execute_call_info
+                        .expect("Failed to get execution data from method")
+                        .execution
+                        .retdata;
+                    let contract_address = return_data
+                        .0
+                        .get(0)
+                        .expect("Failed to get contract_address from return_data");
+                    let contract_address = Felt252::from_bytes_be(contract_address.bytes());
+
+                    insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
+                    insert_value_to_cellref!(vm, panic_data_start, Felt252::from(0))?;
+                    insert_value_to_cellref!(vm, panic_data_end, Felt252::from(0))?;
+                }
+                Result::Err(e) => {
+                    if let TransactionExecutionError::ExecutionError(
+                        EntryPointExecutionError::VirtualMachineExecutionErrorWithTrace {
+                            source,
+                            trace,
+                        },
+                    ) = e
+                    {
+                        let msg = source.to_string();
+                        let unparseable_msg = format!("Deploy failed, full error message: {msg}");
+                        let unparseable_msg = unparseable_msg.as_str();
+
+                        let extracted_panic_data =
+                            try_extract_panic_data(&msg).expect(unparseable_msg);
+                        let mut ptr = vm.add_memory_segment();
+                        insert_value_to_cellref!(vm, panic_data_start, ptr)?;
+
+                        for datum in extracted_panic_data {
+                            insert_at_pointer(vm, &mut ptr, datum);
+                        }
+
+                        insert_value_to_cellref!(vm, deployed_contract_address, contract_address)?;
+                        insert_value_to_cellref!(vm, panic_data_end, ptr)?;
```

Comment:
Why are we inserting an address and panic data end here? Shouldn't it be panic data start end and?

Reply:
Keep in mind this will change anyways after merging #2163 - we will just return plain `Span<felt252>` every time and extract `panic_data` in cairo code. You can look at the [draft](https://github.com/software-mansion-labs/cairo/pull/93/files?diff=split&w=0#diff-954fd47e9dc8a1c411a344ddb12d510aa0b387d21089cae116025ee665cb27faR63-R78) for handling errors in `deploy` cheatcode, but feel free to do it however you feel like is the best

Reply:
One more thing - we should try to not change the user interface too much